packages/superset-ui-chart/src/registries/ChartMetadataRegistrySingleton.ts

```diff
@@ -1,6 +1,7 @@
 import { Registry, makeSingleton, OverwritePolicy } from '@superset-ui/core';
+import { ChartMetaDataConfig } from '../models/ChartMetadata';
 
-class ChartMetadataRegistry extends Registry {
+class ChartMetadataRegistry extends Registry<ChartMetaDataConfig, ChartMetaDataConfig> {
```

Comment:
Should this be `Registry<ChartMetadata>`?

Reply:
`ChartMetadata` gives errors because most calls do not pass a full `ChartMetaData` object, this is a `Partial` for what's required and is used in / matches the constructor.

Reply:
I still think it should be `Registry<ChartMetadata, ChartMetadata>` because 

* There are some additional functions that the `ChartMetadata` provides so it would be nice to ensure that the values in this registry are always the instance of this class. Otherwise devs have to check again after getting an item from registry.
* Most call I believe pass the full `ChartMetadata` instance. The only place things are registered to this registry currently is in the `ChartPlugin` file which always pass the `ChartMetadata` instance

```ts
// in `ChartPlugin.ts`
getChartMetadataRegistry().registerValue(key, this.metadata);
```

* Another place where I see it being registered is the calls in `packages/superset-ui-chart/test/clients/ChartClient.test.ts` which pass plain objects but can easily be updated to make them `ChartMetadata` instances.

Reply:
really good point, updated 👍